test/e2e_tests/specs/Block/block.spec.ts

```diff
@@ -0,0 +1,344 @@
+          await Promise.all([setupTeamForUser(api, userA, 'A'), setupTeamForUser(api, userB, 'B')]);
+        });
+
+        if (connectUsersViaAPI) {
+          await api.connectUsers(userA, userB);
+        }
+
+        await userAPageManager.openMainPage();
+        await userBPageManager.openMainPage();
+
+        await test.step('Preconditions: Signing in User A and User B', async () => {
+          await Promise.all([loginAndSetup(userA, userAPageManager), loginAndSetup(userB, userBPageManager)]);
+        });
+      },
+    );
+
+    test.afterEach(async ({api}) => {
+      await tearDownAll(api);
+    });
+
+    test(
+      'I want to cancel blocking a 1on1 conversation from conversation list 0',
+      {tag: ['@TC-137', '@regression']},
+      async ({pageManager: userAPageManager}) => {
+        const {pages: userAPages, modals: userAModals} = userAPageManager.webapp;
+
+        // Step 1: User A opens conversation with User B
+        await userAPages.conversationList().openConversation(userB.fullName);
+        // Step 2: User A opens the options menu for user B
+        await userAPages.conversationList().clickConversationOptions(userB.fullName);
+        // Step 3: User A opens modal and clicks 'Block' button
+        await userAPages.conversationList().clickBlockConversation();
+        // Step 4: User A clicks 'Cancel' button
+        await userAModals.blockWarning().clickCancel();
+        // Step 5: Conversation is still present, and User A can open it
+        await userAPages.conversationList().openConversation(userB.fullName);
+      },
+    );
+
+    test(
+      'Verify you can block a user who is not in your team',
+      {tag: ['@TC-140', '@regression']},
+      async ({pageManager: userAPageManager, userBPageManager, api}) => {
+        const {pages: userAPages, modals: userAModals} = userAPageManager.webapp;
+        const {pages: userBPages} = userBPageManager.webapp;
+
+        // Step 1: User A and B have a 1:1 conversation
+        await userAPages.conversationList().openConversation(userB.fullName);
+        // Step 2: User A opens conversation info
+        await userAPages.conversation().clickConversationInfoButton();
+        // Step 3: User A clicks "Block conversation" button
+        await userAPages.conversationDetails().clickBlockConversationButton();
+        // Step 4: User A clicks 'Confirm' button
+        await userAModals.blockWarning().clickBlock();
+        // Step 5: User A cannot send message to User B
+        await userAPages.conversationList().openConversation(userB.fullName);
+        await expect(userAPages.conversation().messageInput).not.toBeVisible();
+        // Step 6: User B cannot send message to User A
+        await userBPages.conversationList().openConversation(userA.fullName);
+        await userBPages.conversation().sendMessage('Message after block');
+        expect(await userAPages.conversation().messageCount()).toBe(0);
```

Comment:
Hint: A more reliable way of checking the number of messages would be:
```suggestion
        await expect(userAPages.conversation().messages).toHavecount(0);
```